File: katas/leetcode/1281.Subtract_the_Product_and_Sum_of_Digits_of_an_Integer.py
class Solution:
    def subtractProductAndSum(self, n: int) -> int:
        my_prod = 1
        my_sum = 0

        while n:
            n, remainder = divmod(n, 10)
            my_prod *= remainder
            my_sum += remainder

        return my_prod - my_sum

File: katas/leetcode/1281.Subtract_the_Product_and_Sum_of_Digits_of_an_Integer.java
class Solution {
    public int subtractProductAndSum(int n) {
        int sum = 0;
        int prod = 1;

        while (n != 0) {
            int remainder = n % 10;
            n = n / 10;

            sum += remainder;
            prod *= remainder;
        }

        return prod - sum;
    }
}
